host/sgx/calls.c

```diff
@@ -477,6 +477,8 @@ int __oe_dispatch_ocall(
 
         if (enclave->simulate)
         {
+            oe_mutex_lock(&enclave->lock);
```

Comment:
It is not clear why locking is needed here.
The ocall is dispatched in the same thread that made the ecall and no two threads concurrently use the same tcs.

Reply:
Note: With the newer implementation of oe_enter_sim, the[ GS/FS register restoration](https://github.com/openenclave/openenclave/blob/00bd2d52ea73fe6245e7164ecef6a2a704a9e345/host/sgx/enter.c#L226) is correctly done by the simulation of the ENCLU instruction. This special casing for simulation mode is no longer needed.

Reply:
Good catch Anand. I removed conservative locking.